crypto/txscript/src/opcodes/mod.rs

```diff
@@ -945,10 +942,7 @@ opcode_list! {
             match vm.script_source {
                 ScriptSource::TxInput{tx, ..} => {
                     let [idx]: [i32; 1] = vm.dstack.pop_items()?;
-                    if !(0..=u8::MAX as i32).contains(&idx) {
-                        return Err(TxScriptError::InvalidIndex(idx as usize, tx.inputs().len()))
-                    }
-                    let idx = idx as usize;
+                    let idx = usize::try_from(idx).map_err(|_| TxScriptError::InvalidIndex(idx as usize, tx.inputs().len()))?;
```

Comment:
it would be breacking change. i hope it will be clear to caller since in case of negative it will be very big value.
do you think if i make index i128? to cover both u64 and i64 values?

Reply:
commit df7c2430b2c771c99fbb5776bc300d2a7fc788b1